proposals/incubation/compact.md

```diff
@@ -0,0 +1,305 @@
+
+### Ledger ADTs and public state updates
+Ledger ADTs have both an off-chain and an on-chain data representation, and their operations have both off-chain and on-chain computational implementations.
+In the Midnight Network we have chosen to make both off-chain and on-chain parts the same.
+The Midnight-specific Compact runtime actually uses the *same* representation as the Midnight Network does for ledger ADTs,
+and it actually uses the Impact VM off chain to implement ledger operations.
+
+However, it is not necessary that Compact uses the same implementation off and on chain.
+The Compact compiler itself simply needs to have a backend-specific implementation of off-chain code to use for the ledger,
+and a backend-specific implementation of the on-chain code to be deployed in a contract or else submitted in a transaction, depending on how the backend blockchain works.
+
+## Open Language Design
+The open-source Compact repository will host language feature design documents.
+The Midnight Network has a process for proposing changes called the Midnight Improvement Proposal (MIP) process (The Midnight Foundation, 2025).
+The Compact project will use this process or a suitable variation of it to conduct language feature design in the open,
+and to allow community-driven language feature design.
+
+An example of a similar process being used successfully for programming languages can be found in the Python language community's
+[Python Enhancement Proposal (PEP)](https://peps.python.org/) process.
+
+# Effort and Resources
+
+Given the imminent release of Compact 1.0 as the smart contract language for Midnight,
+Compact should be considered an incubation project because the core concepts are production-ready for at least one chain.
+Midnight has an extensive network of partners actively developing solutions in Compact in preparation for the Midnight launch.
+We recognize that the next phases of Compact’s evolution for cross-blockchain success requires strong governance - not just technical development.
+It is for this reason that we are bringing the released version Compact 1.0 to the Linux Foundation.
+
+It is acknowledged that the cross-chain aspirations will require some research which is beyond the scope of Shielded Technologies and it is our hope that through the LFDT we can collaborate across the blockchain ecosystem to standardize the ways in which programmable privacy for blockchains is accomplished. 
+
+A high level estimate of effort and resources based on current state of compact is as follows:
+- **69,000+ lines** of mature Scheme compiler code
+- **Production-ready** with comprehensive CI/CD (17+ GitHub workflows)
+- **Version 0.24.104** with active development preparing for 1.0 release in Q3
+- **Nanopass architecture** - excellent foundation for multi-target compilation
+
+## Major Areas of Focus
+- Cross-blockchain abstraction layer
+- Multi-target compiler backends
+- Cross-chain security
```

Comment:
Please provide additional context, what does `cross-chain security` mean per the project proposal?